salt/modules/zypperpkg.py

```diff
@@ -1744,7 +1807,11 @@ def upgrade(
         Specify a list of package repositories to upgrade from. Default: None
 
     novendorchange
-        If set to True, no allow vendor changes. Default: False
+        DEPRECATED(use allowvendorchange): If set to True, do not allow vendor changes. Default: True
```

Comment:
**question** Do we intend to actually remove this? If we do then we should add a `warn_until` if `novendorchange` is different. It actually might be a good idea to set the default to None in the function call, so...

```
def upgrade(..., novendorchange=None,...):
    if novendorchange is not None:
        warn_until(..., "novendorchange is deprecated and will be removed after the XXX release.")
    else:
       novendorchange = not allowvendorchange
```

See https://archive.repo.saltproject.io/en/2015.5/topics/development/deprecations.html for more info.

Reply:
Well, I'd recommend getting rid of `novenderchange`.   It was the wrong option from the beginning.

Reply:
If it's completely broken (i.e. it's a bug) then that's fine, but even if it's a nonsense functionality we still don't want to just break things for people who may be relying on it. Or just using it.

Reply:
It never worked from the beginning.  But I'm not saying it needs to be removed right this second.

Reply:
My suggestion then, is to go ahead and use `warn_until` - can even mention something in the warning about how `novendorchange` never worked right or whatever.